### PR TITLE
add timeout

### PR DESCRIPTION
**Pull request by resession**
i added a timeout for all ipfs queries.

### CODE REVIEW

**Comment by resession** [comment]
added a timeout of 2 minutes for ipns

File: index.js [hunk 1]
```diff
@@ -22,7 +24,7 @@ module.exports = function makeIPFSFetch ({ ipfs }) {
     headers.Allow = SUPPORTED_METHODS.join(', ')
 
     async function getStat (path) {
-      return exporter(path, ipfs.block, { signal, preload: false })
+      return exporter(path, ipfs.block, { signal, preload: false, ipfsTimeout })
```
Comment (RangerMauve) [inline index.js]:
```suggestion
      return exporter(path, ipfs.block, { signal, preload: false, timeout: ipfsTimeout })
```

Should all of these be changed to `timeout: ipfsTimeout`? I think that's what it says in the docs.

Also would you be able to add a unit test to make sure the timeout actually works?

**Review by RangerMauve — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (resession) [reply index.js]: looks like i got used to copy and pasting lol, fixed now. i will go ahead. i will work on the unit tests now.

Comment (resession) [reply index.js]: made the test and tested on a CID that is offline and it timeouts like it should

**Comment by resession** [comment]
let me know any feedback

File: test.js [hunk 2]
```diff
@@ -63,6 +63,18 @@ test('Load a file via fetch', async (t) => {
     const text = await response.text()
 
     t.equal(text, TEST_DATA, 'Got expected file content')
+
+    const testTimeout = await fetch('ipfs://QmYCmKG1uJhimRqrjusTbwfGRhfbcKpTh2SKDBVwZFRC5e')
+
+    t.ok(testTimeout, 'Got a response object')
+    t.equal(testTimeout.status, 500, 'Got OK in response')
```
Comment (RangerMauve) [inline test.js]: Is the wording here accurate? Should this say `Got Error in response`?

File: test.js [hunk 3]
```diff
@@ -690,6 +702,37 @@ test('Resolve IPNS', async (t) => {
...
+    ipfs = await getInstance()
+
+    const fetch = await makeIPFSFetch({ ipfs })
+
+    t.pass('Able to make create fetch instance')
+
+    const testTimeout = await fetch('ipfs://QmYCmKG1uJhimRqrjusTbwfGRhfbcKpTh2SKDBVwZFRC5e')
+
+    t.ok(testTimeout, 'Got a response object')
+    t.equal(testTimeout.status, 500, 'Got OK in response')
```
Comment (RangerMauve) [inline test.js]: Should this say "Got error in response"?

Also, would it be possible to return a [408 status](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/408) if it times out?

**Review by RangerMauve — CHANGES_REQUESTED** [review]

File: test.js (hunk 2 above)
Comment (RangerMauve) [reply test.js]: Also is this section necessary given the other test on like 700?

Comment (resession) [reply test.js]: > Is the wording here accurate? Should this say `Got Error in response`?

fixed

> Also is this section necessary given the other test on like 700?

i was trying to follow a similar pattern to the other tests, i can take this part out if its not needed, by the way, what are you referring to when you say 700?

File: test.js (hunk 3 above)
Comment (resession) [reply test.js]: fixed

**Review by RangerMauve — APPROVED** [review]
Looks perfect, thank you!

File: test.js (hunk 2 above)
Comment (RangerMauve) [reply test.js]: By 700 I meant line 700. 😅 What you have now is perfect!

**Pull request merged**